### PR TITLE
Fix issue with Swagger UI showing only southbound routes.

### PR DESCRIPTION
**Pull request by brianparry**
* Chain swagger registration promises to avoid race condition.

Resolves https://github.com/RackHD/on-http/issues/444

### CODE REVIEW

**Comment by benbp** [comment]
Out of curiosity, why does doing them in sequence fix it rather than just settling on resolution of all promises?

**Comment by brianparry** [comment]
@benbp The problem is caused by the fact that we are assigning two swagger files to the same basepath.  Swagger middleware will return a swagger document corresponding to the first config to be registered. If monorail-2.0-sb.yaml is registered first, you'll see only those routes in the UI.  This is really just a workaround until we get to this story: https://www.pivotaltracker.com/n/projects/1496834/stories/129367887

**Comment by heckj** [comment]
👍

**Comment by BillyAbildgaard** [comment]
good job finding this bug, Brian!

+1

**Comment by yyscamper** [comment]
Good finding, :+1:

File: lib/services/http-service.js [hunk 1]
```diff
@@ -169,27 +169,32 @@ function httpServiceFactory(
...
+            .then(function(swaggerExpress) {
+                app.use(require('restify-links')());
+                return swaggerExpress.register(app);
+            });
+        })
+        .then(function() {
+            return swaggerCreateAsync(sbSwaggerConfig)
+            .then(function(swaggerExpress) {
+                return swaggerExpress.register(app);
+            })
```
Comment (yyscamper) [inline lib/services/http-service.js]: Cheer, The houndci recoveries ?!

**Pull request merged**